daemon/start.go

```diff
@@ -236,6 +239,19 @@ func (daemon *Daemon) containerStart(ctx context.Context, daemonCfg *configStore
 		}
 	}()
 
+	if runtime.GOOS == "linux" && !container.Config.NetworkDisabled {
+		nspath, ok := oci.NamespacePath(spec, specs.NetworkNamespace)
+		if ok && nspath == "" { // the runtime has been instructed to create a new network namespace for tsk.
+			sb, err := daemon.netController.GetSandbox(container.ID)
+			if err != nil {
+				return errdefs.System(err)
+			}
+			if err := sb.SetKey(fmt.Sprintf("/proc/%d/ns/net", tsk.Pid())); err != nil {
```

Comment:
Oops, that line doesn't compile on Windows. I can fix that...

Reply:
Oh, I've done it by adding a `SetKey()` to `sandbox_unsupported.go` and fixing up the commit - hopefully that's what you planned? If not I can do it differently.

Reply:
https://github.com/corhere/moby/commit/4b0cf3d6f3589486cee969ef882ea8323bf48a30

I would prefer not to define a stub `SetKey` for Windows if it can be avoided. It can't do anything meaningful with the argument so can never be anything but a no-op stub. And there's already more than enough [interfaces in libnetwork which are only meaningful in one context that have to be stubbed out by all other implementations.](https://en.wikipedia.org/wiki/Interface_segregation_principle)

Reply:
Ok, thank you ... I've swapped it in.

(Just had to fix a conflict in `daemon/oci_linux.go`, the list of opts changed just after your rebase.)